webapi/src/main/scala/org/knora/webapi/routing/admin/UsersRouteADM.scala

```diff
@@ -121,108 +129,71 @@ class UsersRouteADM(routeData: KnoraRouteData) extends KnoraRoute(routeData) wit
-                }
-            } ~
-                delete {
-                    /* delete a user identified by iri */
-                    requestContext => {
-                        val userIri = stringFormatter.validateAndEscapeIri(value, throw BadRequestException(s"Invalid user IRI $value"))
-
-                        if (userIri.equals(KnoraSystemInstances.Users.SystemUser.id) || userIri.equals(KnoraSystemInstances.Users.AnonymousUser.id)) {
-                            throw BadRequestException("Changes to built-in users are not allowed.")
-                        }
-
-                        /* update existing user's status to false */
-                        val requestMessage: Future[UserChangeStatusRequestADM] = for {
-                            requestingUser <- getUserADM(requestContext)
-                        } yield UserChangeStatusRequestADM(
-                            userIri,
-                            changeUserRequest = ChangeUserApiRequestADM(status = Some(false)),
-                            requestingUser,
-                            apiRequestID = UUID.randomUUID()
-                        )
-
-                        RouteUtilADM.runJsonRoute(
-                            requestMessage,
-                            requestContext,
-                            settings,
-                            responderManager,
-                            log
-                        )
-                    }
-                }
-        } ~
-        path("admin" / "users" / "projects" / Segment) { userIri =>
-            get {
-                /* get user's project memberships */
+    /**
+      *  return a single user identified by email
+      */
+    private def getUserByEmail: Route = path("admin" / "users" / "email" / Segment) { value =>
+        get {
+            requestContext =>
+                val requestMessage: Future[UserGetRequestADM] = for {
+                    requestingUser <- getUserADM(requestContext)
+                    userEmail = stringFormatter.validateEmailAndThrow(value, throw BadRequestException(s"Invalid email $value"))
+                } yield UserGetRequestADM(UserIdentifierADM(email = Some(userEmail)), UserInformationTypeADM.RESTRICTED, requestingUser)
+
+                RouteUtilADM.runJsonRoute(
+                    requestMessage,
+                    requestContext,
+                    settings,
+                    responderManager,
+                    log
+                )
+        }
+    }
+
+    /**
+      *  return a single user identified by username
+      */
+    private def getUserByUsername: Route = path("admin" / "users" / "username" / Segment) { value =>
+        get {
+            requestContext =>
+                val requestMessage: Future[UserGetRequestADM] = for {
+                    requestingUser <- getUserADM(requestContext)
+                } yield UserGetRequestADM(UserIdentifierADM(username = Some(value)), UserInformationTypeADM.RESTRICTED, requestingUser)
```

Comment:
The username needs to be escaped with `StringFormatter.toSparqlEncodedString()`.